src/components/FeaturePanel/PublicTransport/PublicTransport.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useState, useEffect } from 'react';
+import { useRouter } from 'next/router';
+import { requestLines } from './requestRoutes';
+import { PublicTransportWrapper } from './PublicTransportWrapper';
+import { FeatureTags } from '../../../services/types';
+import { LineNumber } from './LineNumber';
+
+interface PublicTransportProps {
+  tags: FeatureTags;
+}
+
+export const PublicTransport: React.FC<PublicTransportProps> = ({ tags }) => {
+  const isPublicTransport =
+    Object.keys(tags).includes('public_transport') ||
+    tags.railway === 'station';
+
+  if (!isPublicTransport) {
+    return null;
+  }
+
+  const [data, setData] = useState(null);
+  const [loading, setLoading] = useState(true);
+  const router = useRouter();
+
+  useEffect(() => {
+    const loadData = async () => {
+      const lines = await requestLines(
+        router.query.all[0] as any,
+        Number(router.query.all[1]),
+      );
+      setData(lines);
+      setLoading(false);
+    };
+
+    loadData();
+  }, []);
+
+  return (
+    <div>
+      {loading ? (
+        <div>...</div>
```

Comment:
Could you please add "animated three dots loader" like [here](https://github.com/zbycz/osmapp/blob/master/src/components/FeaturePanel/ObjectsAround.tsx#L118-L120)? It is much more clear for the user, when the loading is eg. stuck forever, etc.

Since this is shown only for Public transport entities, it could be perhaps shown even with the "Loading" string, or even better UX could be "Loading line numbers...", what do you think? When there is no line number, some text like "No lines number found for this station.", perhaps with link to overpass?

Reply:
I already wanted something like "Load routes..." but I don't know how internationalization is handled in this project.